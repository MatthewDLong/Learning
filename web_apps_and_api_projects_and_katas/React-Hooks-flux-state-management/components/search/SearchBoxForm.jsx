import React from "react";
import SearchBoxHeading from "./SearchBoxHeading";
import SearchBoxLabel from "./SearchBoxLabel";
import SearchBoxInput from "./SearchBoxInput";

import { searchbox__form } from "./search.css";

const SearchBoxForm = () => (
  <form role="search" id="searchbox" className={searchbox__form}>
    <SearchBoxHeading />
    <SearchBoxLabel />
    <SearchBoxInput maxResults={6} />
  </form>
);

export default SearchBoxForm;
