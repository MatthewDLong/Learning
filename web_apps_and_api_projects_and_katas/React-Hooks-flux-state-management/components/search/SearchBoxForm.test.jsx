import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import SearchBoxForm from "./SearchBoxForm";

describe("SearchBoxForm", () => {
  it("renders correctly", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<SearchBoxForm />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
