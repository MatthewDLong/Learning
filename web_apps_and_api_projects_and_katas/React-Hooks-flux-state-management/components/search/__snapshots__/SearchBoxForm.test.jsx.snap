// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBoxForm renders correctly 1`] = `
<form
  className="searchbox__form"
  id="searchbox"
  role="search"
>
  <SearchBoxHeading />
  <SearchBoxLabel />
  <SearchBoxInput
    maxResults={6}
  />
</form>
`;
